### PR TITLE
Check Hidden values

### LINKED ISSUE #91: Ignore hidden values in modules
It can happen that modules have hidden values.

For instance, in the following example:

```
module type M = sig
  val x : int
end

module type N = sig
  include M
  val x : float
end
```
the `x` value coming from `include M` is shadowed by the "redeclaration" of `x`.

The visibility of a signature item is given by the [`visibility`](https://github.com/ocaml/ocaml/blob/trunk/typing/types.mli#L606) value of the [`signature_item`](https://github.com/ocaml/ocaml/blob/trunk/typing/types.mli#L626).

However, the API diffing currently ignores this visibility property.

This issue consists of two steps:

- Exhibit in a test a bug in the diffing that is due to this "visibility oversight"
- Fix it by taking the visibility into account when diffing two API.

### PR DESCRIPTION
**Pull request by NchamJosephMuam**
closes #91 
Hello @NathanReb @panglesd 
I added some code to explicitly check for visibility. I will now continue with writing the tests.

### CODE REVIEW

File: lib/diff.ml [hunk 1]
```diff
@@ -38,18 +38,28 @@ and sig_item =
 let extract_items items =
   List.fold_left
     (fun tbl item ->
+      let visibility = Types.item_visibility item in
```
Comment (NathanReb) [inline lib/diff.ml:41]: You don't need to call this function, the visibility is part of each sig item constructor arguments so you can directly pattern match over it.

File: lib/diff.ml [hunk 2]
```diff
@@ -38,18 +38,28 @@ and sig_item =
...
       match item with
       | Sig_module (id, _, mod_decl, _, _) ->
-          Sig_item_map.add ~name:(Ident.name id) Sig_item_map.Module mod_decl
-            tbl
+          if visibility = Types.Hidden then tbl
```
Comment (NathanReb) [inline lib/diff.ml:44]: Same here, you don't need to go through an if-then-else, you can directly match over the `visibility` argument (last one) and let the pattern matching wildcard case at the end handle all `Hidden` items.

File: lib/diff.ml [hunk 3]
```diff
@@ -39,15 +39,15 @@ let extract_items items =
   List.fold_left
     (fun tbl item ->
       match item with
-      | Sig_module (id, _, mod_decl, _, _) ->
+      | Sig_module (id, _, mod_decl, _, Types.Exported) ->
```
Comment (NathanReb) [inline lib/diff.ml:42]: The local open should not be necessary here! Same applies to the other cases.
```suggestion
      | Sig_module (id, _, mod_decl, _, Exported) ->
```

**Review by NathanReb — COMMENTED** [review]
This looks good, thanks! Could you add a changelog entry in the `Fixed` section?

**Comment by NchamJosephMuam** [comment]
> This looks good, thanks! Could you add a changelog entry in the `Fixed` section?
Yeah, thank you for the review. I'll add the changelog in the next commit

**Comment by panglesd** [comment]
Since you merged `main` in this PR, there is a new case to handle in the pattern-match!

**Comment by NchamJosephMuam** [comment]
> Since you merged `main` in this PR, there is a new case to handle in the pattern-match!

Thank you for catching that. I have updated it in the latest commit.

**Review by NathanReb — APPROVED** [review]

**Pull request merged**